### DISCUSSION TITLE
Flashing the bootloader onto the AI Deck

### DISCUSSION THREAD

**Discussion by laurenbramblett** [bitcraze/discussions#635] [Q&A]
Hello Everyone!

I am struggling with the hardware required for flashing the bootloader onto the gap8 on the AI Deck. From what I can tell, I need the following:
1. ARM-USB-TINY-H which typically has a 2 row 20 pin JTAG cable
2. ARM-JTAG-20-10 adapter for taking the 20 pin output from the ARM-USB-TINY-H and converting it to the pins on the AI deck
3. USB-B to USB-A adapter for plugging in to a laptop (I have no idea where to find a female USB-B to USB-A adapter - I cannot find very many for purchase)

I have an AI Deck 1.1 Rev C1 and recently purchased an ST-Link V2 thinking it could be useful, but after a deeper search it seems there could be extra steps involved? Could anyone clarify these steps? If I can get away with only buying the ARM-JTAG-20-10 adapter, that would be great.

Thank you in advance!

Lauren

**Comment by hmllr** [comment]
Hi :)
Unfortunately we haven't managed to flash code to an gap8 with an ST-Link -  which doesn't necessarily mean it cannot work, but that we cannot help with that. 
You only need to flash the bootloader once, so if you don't want to buy one I guess you could try to borrow one in an university close by or such for some minutes - however, an olimex might also be nice to have for debugging easier while developing code for the gap8. 
Best,
Hanna
